packages/protocol/test/governance/election.ts

```diff
@@ -0,0 +1,854 @@
+      assertEqualBN(min, newElectableValidators.min)
+      assertEqualBN(max, newElectableValidators.max)
+    })
+
+    it('should emit the ElectableValidatorsSet event', async () => {
+      const resp = await election.setElectableValidators(
+        newElectableValidators.min,
+        newElectableValidators.max
+      )
+      assert.equal(resp.logs.length, 1)
+      const log = resp.logs[0]
+      assertContainSubset(log, {
+        event: 'ElectableValidatorsSet',
+        args: {
+          min: newElectableValidators.min,
+          max: newElectableValidators.max,
+        },
+      })
+    })
+
+    it('should revert when the minElectableValidators is zero', async () => {
+      await assertRevert(election.setElectableValidators(0, newElectableValidators.max))
+    })
+
+    it('should revert when the min is greater than max', async () => {
+      await assertRevert(
+        election.setElectableValidators(
+          newElectableValidators.max.plus(1),
+          newElectableValidators.max
+        )
+      )
+    })
+
+    it('should revert when the values are unchanged', async () => {
+      await assertRevert(
+        election.setElectableValidators(electableValidators.min, electableValidators.max)
+      )
+    })
+
+    it('should revert when called by anyone other than the owner', async () => {
+      await assertRevert(
+        election.setElectableValidators(newElectableValidators.min, newElectableValidators.max, {
+          from: nonOwner,
+        })
+      )
+    })
+  })
+
+  describe('#setMaxNumGroupsVotedFor', () => {
+    const newMaxNumGroupsVotedFor = maxNumGroupsVotedFor.plus(1)
+    it('should set the max electable validators', async () => {
+      await election.setMaxNumGroupsVotedFor(newMaxNumGroupsVotedFor)
+      assertEqualBN(await election.maxNumGroupsVotedFor(), newMaxNumGroupsVotedFor)
+    })
+
+    it('should emit the MaxNumGroupsVotedForSet event', async () => {
+      const resp = await election.setMaxNumGroupsVotedFor(newMaxNumGroupsVotedFor)
+      assert.equal(resp.logs.length, 1)
+      const log = resp.logs[0]
+      assertContainSubset(log, {
+        event: 'MaxNumGroupsVotedForSet',
+        args: {
+          maxNumGroupsVotedFor: new BigNumber(newMaxNumGroupsVotedFor),
+        },
+      })
+    })
+
+    it('should revert when the maxNumGroupsVotedFor is unchanged', async () => {
+      await assertRevert(election.setMaxNumGroupsVotedFor(maxNumGroupsVotedFor))
+    })
+
+    it('should revert when called by anyone other than the owner', async () => {
+      await assertRevert(
+        election.setMaxNumGroupsVotedFor(newMaxNumGroupsVotedFor, { from: nonOwner })
+      )
+    })
+  })
+
+  describe('#markGroupEligible', () => {
+    const group = accounts[1]
+    describe('when the group has members', () => {
+      beforeEach(async () => {
+        await mockValidators.setMembers(group, [accounts[9]])
+      })
+
+      describe('when the group has no votes', () => {
```

Comment:
Is this "no votes" context significant? If so, should there be a "when the group has votes" `describe`?

Reply:
It's somewhat significant, and eventually yeah it would be good to add a test case when the group has votes but I don't think it's critical right now